### ISSUE TITLE
Update Valve Sensor Faild - EOF occurred in violation of protocol

### ISSUE THREAD

**Issue by Gobolin80** [albertogeniola/meross-homeassistant#285]
**Describe the bug**
Hi,
since few days in the logs this error appears.
Today I have 2400 occurences.
In my opinion something changes in meross backend.

Thanks for your job

**Your environment**
HomeAssistant version: 2021.11.3
Hassio Version (if applicable): core
Hardware environment: Raspberry Pi 4 + 2 Meross Valve V3 + 1 Meross Valve V2

**Logs taken when the issue happened**
```
Logger: homeassistant.helpers.entity
Source: custom_components/meross_cloud/sensor.py:296
Integration: Meross Cloud IoT
First occurred: 13:37:44 (2346 occurrences)
Last logged: 18:31:00

Update for sensor.termo_cucina_mts100v3_temperature_sensor_degc_0 fails
Update for sensor.termo_bagno_mts100v3_temperature_sensor_degc_0 fails
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 468, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 658, in async_device_update
    raise exc
  File "/config/custom_components/meross_cloud/sensor.py", line 296, in async_update
    await self._device.async_get_temperature()
  File "/usr/local/lib/python3.9/site-packages/meross_iot/controller/known/subdevice.py", line 235, in async_get_temperature
    res = await self._hub._execute_command(method="GET", namespace=Namespace.HUB_MTS100_TEMPERATURE,
  File "/usr/local/lib/python3.9/site-packages/meross_iot/controller/device.py", line 281, in _execute_command
    return await self._manager.async_execute_cmd(destination_device_uuid=self.uuid,
  File "/usr/local/lib/python3.9/site-packages/meross_iot/manager.py", line 838, in async_execute_cmd
    return await self.async_execute_cmd_client(client=client,
  File "/usr/local/lib/python3.9/site-packages/meross_iot/manager.py", line 903, in async_execute_cmd_client
    response = await self._async_send_and_wait_ack(
  File "/usr/local/lib/python3.9/site-packages/meross_iot/manager.py", line 915, in _async_send_and_wait_ack
    message_info = client.publish(
  File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 1257, in publish
    rc = self._send_publish(
  File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 2693, in _send_publish
    return self._packet_queue(PUBLISH, packet, mid, qos, info)
  File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 3016, in _packet_queue
    return self.loop_write()
  File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 1577, in loop_write
    rc = self._packet_write()
  File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 2464, in _packet_write
    write_length = self._sock_send(
  File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 649, in _sock_send
    return self._sock.send(buf)
  File "/usr/local/lib/python3.9/ssl.py", line 1173, in send
    return self._sslobj.write(data)
ssl.SSLEOFError: EOF occurred in violation of protocol (_ssl.c:2483)
```

**Comment by talik31** [comment]
Hello, first thing first : a huge bravo and a big thank you to the developer of this integration :)

The integration was working quite fine for the last year, only a rare few disconnect but around 2 weeks ago the integration gone wild.

I'm also facing the same issue :
##########
Cette erreur provient d'une intégration personnalisée

Logger: homeassistant.helpers.entity
Source: custom_components/meross_cloud/sensor.py:330
Integration: Meross Cloud IoT
First occurred: 14 novembre 2021, 00:59:02 (9440 occurrences)
Last logged: 00:26:00

Update for sensor.prise_lampe_xxxxxx_mss310_power_sensor_w_0 fails
Update for sensor.prise_sdb_mss310_power_sensor_w_0 fails
Update for sensor.prise_lit_mss310_power_sensor_w_0 fails
Update for sensor.ventilateur_mss310_power_sensor_w_0 fails
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 468, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 658, in async_device_update
    raise exc
  File "/config/custom_components/meross_cloud/sensor.py", line 330, in async_update
    await self._device.async_get_instant_metrics(channel=self._channel_id)
  File "/usr/local/lib/python3.9/site-packages/meross_iot/controller/mixins/electricity.py", line 43, in async_get_instant_metrics
    result = await self._execute_command(method="GET",
  File "/usr/local/lib/python3.9/site-packages/meross_iot/controller/device.py", line 281, in _execute_command
    return await self._manager.async_execute_cmd(destination_device_uuid=self.uuid,
  File "/usr/local/lib/python3.9/site-packages/meross_iot/manager.py", line 838, in async_execute_cmd
    return await self.async_execute_cmd_client(client=client,
  File "/usr/local/lib/python3.9/site-packages/meross_iot/manager.py", line 903, in async_execute_cmd_client
    response = await self._async_send_and_wait_ack(
  File "/usr/local/lib/python3.9/site-packages/meross_iot/manager.py", line 915, in _async_send_and_wait_ack
    message_info = client.publish(
  File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 1257, in publish
    rc = self._send_publish(
  File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 2693, in _send_publish
    return self._packet_queue(PUBLISH, packet, mid, qos, info)
  File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 3016, in _packet_queue
    return self.loop_write()
  File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 1577, in loop_write
    rc = self._packet_write()
  File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 2464, in _packet_write
    write_length = self._sock_send(
  File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 649, in _sock_send
    return self._sock.send(buf)
  File "/usr/local/lib/python3.9/ssl.py", line 1173, in send
    return self._sslobj.write(data)
ssl.SSLEOFError: EOF occurred in violation of protocol (_ssl.c:2483)
##########


More info about my system :
## System Health

version | core-2021.11.3
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.9.7
os_name | Linux
os_version | 5.10.17-v8
arch | aarch64
timezone | Europe/Paris

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
Github API Calls Remaining | 5000
Installed Version | 1.17.1
Stage | running
Available Repositories | 900
Installed Repositories | 19

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | false
-- | --
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Home Assistant OS 6.6
-- | --
update_channel | stable
supervisor_version | supervisor-2021.10.8
docker_version | 20.10.8
disk_total | 57.8 GB
disk_used | 11.0 GB
healthy | true
supported | true
board | rpi4-64
supervisor_api | ok
version_api | ok
installed_addons | File editor (5.3.3), SSH & Web Terminal (9.1.0), ESPHome (2021.10.3), AirCast (3.2.0), Let's Encrypt (4.11.0), NGINX Home Assistant SSL proxy (3.0.2), RPC Shutdown (2.2), Mosquitto broker (6.0.1)

</details>

<details><summary>Lovelace</summary>

dashboards | 1
-- | --
resources | 9
views | 11
mode | storage

</details>

**Comment by nchieffo** [comment]
Similar happened to me

`This error originated from a custom integration.

Logger: homeassistant.helpers.entity
Source: custom_components/meross_cloud/sensor.py:401
Integration: Meross Cloud IoT
First occurred: 7:44:23 PM (724 occurrences)
Last logged: 8:44:38 PM

Update for sensor.presa_scrivania_studio_mss310_power_sensor_v_0 fails
Update for sensor.presa_tapparelle_mss310_power_sensor_v_0 fails
Update for sensor.presa_stufetta_bagno_mss310_power_sensor_v_0 fails
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 468, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 658, in async_device_update
    raise exc
  File "/config/custom_components/meross_cloud/sensor.py", line 401, in async_update
    await self._device.async_get_instant_metrics(channel=self._channel_id)
  File "/usr/local/lib/python3.9/site-packages/meross_iot/controller/mixins/electricity.py", line 43, in async_get_instant_metrics
    result = await self._execute_command(method="GET",
  File "/usr/local/lib/python3.9/site-packages/meross_iot/controller/device.py", line 281, in _execute_command
    return await self._manager.async_execute_cmd(destination_device_uuid=self.uuid,
  File "/usr/local/lib/python3.9/site-packages/meross_iot/manager.py", line 838, in async_execute_cmd
    return await self.async_execute_cmd_client(client=client,
  File "/usr/local/lib/python3.9/site-packages/meross_iot/manager.py", line 903, in async_execute_cmd_client
    response = await self._async_send_and_wait_ack(
  File "/usr/local/lib/python3.9/site-packages/meross_iot/manager.py", line 915, in _async_send_and_wait_ack
    message_info = client.publish(
  File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 1257, in publish
    rc = self._send_publish(
  File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 2693, in _send_publish
    return self._packet_queue(PUBLISH, packet, mid, qos, info)
  File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 3016, in _packet_queue
    return self.loop_write()
  File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 1577, in loop_write
    rc = self._packet_write()
  File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 2464, in _packet_write
    write_length = self._sock_send(
  File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 649, in _sock_send
    return self._sock.send(buf)
  File "/usr/local/lib/python3.9/ssl.py", line 1173, in send
    return self._sslobj.write(data)
ssl.SSLError: [SSL: BAD_LENGTH] bad length (_ssl.c:2483)
`

**Comment by Lazza** [comment]
I get the same issue. Is there any other kind of debug information that we can share?

Dev, please just ask if you need more info and I will provide it. 😄 Thank you for the integration.

**Comment by toot** [comment]
I switched to [Meross LAN](https://github.com/krahabb/meross_lan), this integration works locally and is absolutely reliable.
Tbh i should have it done earlier.

**Comment by albertogeniola** [comment]
Addressed in #318 . Closing.
